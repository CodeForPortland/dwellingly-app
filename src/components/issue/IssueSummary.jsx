import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../icon/Icon';

const IssueSummary = ({ issue, editSummary, urgency, issueNote }) => (
  <div>
    <div className="newIssueSummaryInfo padding--1em">
      <span className="newIssueTitle title">{issue}</span>
      <span
        className="newIssueSummaryEdit"
        onClick={editSummary}
        role="presentation"
        id="issueType">
        <Icon icon="pencil" />
      </span>
    </div>
    <div className="newIssueSummaryInfo padding--1em">
      <div className="newIssueTitle">
        <span className="title">Urgency Level</span>
        <span className={`status status--${urgency}`}>{urgency}</span>
      </div>
      <span
        className="newIssueSummaryEdit"
        onClick={editSummary}
        role="presentation"
        id="urgency">
        <Icon icon="pencil" />
      </span>
    </div>
    <div className="newIssueSummaryInfo padding--1em">
      <span className="newIssueTitle msgbox">
        {issueNote.split('\n').map(i => <p>{i}</p>)}
      </span>
      <span
        className="newIssueSummaryEdit"
        onClick={editSummary}
        role="presentation"
        id="attachments">
        <Icon icon="pencil" />
      </span>
    </div>
  </div>
);

IssueSummary.propTypes = {
  issue: PropTypes.string.isRequired,
  editSummary: PropTypes.func.isRequired,
  urgency: PropTypes.string.isRequired,
  issueNote: PropTypes.string.isRequired
};

export default IssueSummary;
