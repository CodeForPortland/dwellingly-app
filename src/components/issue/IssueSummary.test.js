import React from 'react';
import renderer from 'react-test-renderer';
import IssueSummary from './IssueSummary';

describe('IssueSummary Component', () => {
  it('renders correctly when issueNote prop exists', () => {
    const props = {
      issueNote: '',
    }
    const IssueSummaryComponent = renderer.create(<IssueSummary {...props}/>).toJSON();
    expect(IssueSummaryComponent).toMatchSnapshot();
  });
  it('renders issue text passed down as props', () => {
    const props = {
      issue: 'property damage',
      issueNote: '',
    }
    const IssueSummaryComponent = renderer.create(<IssueSummary {...props}/>).toJSON();
    expect(IssueSummaryComponent).toMatchSnapshot();
  });
  it('renders urgency className and text passed down as props', () => {
    const props = {
      urgency: 'high',
      issueNote: '',
    }
    const IssueSummaryComponent = renderer.create(<IssueSummary {...props}/>).toJSON();
    expect(IssueSummaryComponent).toMatchSnapshot();
  });
});
