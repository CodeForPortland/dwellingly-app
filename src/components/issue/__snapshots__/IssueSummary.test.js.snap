// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSummary Component renders correctly when issueNote prop exists 1`] = `
<div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <span
      className="newIssueTitle title"
    />
    <span
      className="newIssueSummaryEdit"
      id="issueType"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <div
      className="newIssueTitle"
    >
      <span
        className="title"
      >
        Urgency Level
      </span>
      <span
        className="status status--undefined"
      />
    </div>
    <span
      className="newIssueSummaryEdit"
      id="urgency"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <span
      className="newIssueTitle msgbox"
    >
      <p>
        
      </p>
    </span>
    <span
      className="newIssueSummaryEdit"
      id="attachments"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;

exports[`IssueSummary Component renders issue text passed down as props 1`] = `
<div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <span
      className="newIssueTitle title"
    >
      property damage
    </span>
    <span
      className="newIssueSummaryEdit"
      id="issueType"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <div
      className="newIssueTitle"
    >
      <span
        className="title"
      >
        Urgency Level
      </span>
      <span
        className="status status--undefined"
      />
    </div>
    <span
      className="newIssueSummaryEdit"
      id="urgency"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <span
      className="newIssueTitle msgbox"
    >
      <p>
        
      </p>
    </span>
    <span
      className="newIssueSummaryEdit"
      id="attachments"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;

exports[`IssueSummary Component renders urgency className and text passed down as props 1`] = `
<div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <span
      className="newIssueTitle title"
    />
    <span
      className="newIssueSummaryEdit"
      id="issueType"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <div
      className="newIssueTitle"
    >
      <span
        className="title"
      >
        Urgency Level
      </span>
      <span
        className="status status--high"
      >
        high
      </span>
    </div>
    <span
      className="newIssueSummaryEdit"
      id="urgency"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="newIssueSummaryInfo padding--1em"
  >
    <span
      className="newIssueTitle msgbox"
    >
      <p>
        
      </p>
    </span>
    <span
      className="newIssueSummaryEdit"
      id="attachments"
      role="presentation"
    >
      <span
        className="icon icon--pencil"
      >
        <svg
          viewBox="0 0 50 50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;
